### DISCUSSION TITLE
Create new zellij session with specific command/shell

### DISCUSSION THREAD

**Discussion by holly-hacker** [zellij-org/zellij#2236] [Ideas]
I am aware that you can do `zellij r -- <my_command>` to start a new pane in an existing zellij session that runs a command. However, I can't find to open a *new* zellij session that uses a specific shell or runs a specific program. Is this currently implemented?

**Comment by mdrv** [comment]
I’m also waiting for this feature. Currently, the only way I found is to specify a `.kdl` layout file with predefined command(s) and execute `zellij -l test.kdl`. You can find more info here:

https://zellij.dev/documentation/layouts.html

However, I would like a way that doesn’t need to create a file and have the ability to pass commands as arguments, since having to edit the file first just because you want to execute different commands is quite a hassle.

**Reply by Lockszmith-GH** [reply]
FYI, in bash (and zsh) with a [heredoc and Process Substitution](https://stackoverflow.com/questions/4664229/here-document-as-an-argument-to-bash-function) you can 'inline' the .kdl by running something like the following:

```bash
zellij -l <(cat <<KDL
### KDL layout content here
KDL
)
```

This way, you don't need to save the layout to a file.

**Comment by non-bin** [comment]
Sorry to bump... but bump

I want to be able to ssh to another machine and run a script in zellij, in one line

**Reply by Lockszmith-GH** [reply]
You have **[zellij run](https://zellij.dev/documentation/zellij-run.html)** (or `zr`) or you can create a wrapper function around [the heredoc solution mentioned above](https://github.com/zellij-org/zellij/discussions/2236#discussioncomment-7306374).

**Reply by non-bin** [reply]
I guess this is fine, but I have a somewhat complex layout I like and want to reuse it, without including it in the heredoc. Is there a way to import other layout files?

Otherwise `zellij --command <something>` or `zellij r --new -- something` would be useful to run a command in the default focused pane

**Reply by Lockszmith-GH** [reply]
I have a similar situation, to solve that I have a dedicated template layout file that I pass through envsubst, and that layout calls a script.
It isn't adhoc, but it works very well.
If you need the layout to be dynamic, then your script can generate the template.
